src/main/java/randoop/resource/MethodListGen.java

```diff
@@ -0,0 +1,241 @@
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.plumelib.reflection.Signatures;
+import randoop.main.RandoopUsageError;
+import randoop.reflection.EverythingAllowedPredicate;
+import randoop.reflection.VisibilityPredicate;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+import scenelib.annotations.el.AMethod;
+import scenelib.annotations.el.AScene;
+import scenelib.annotations.io.classfile.ClassFileReader;
+
+/**
+ * This tool generates the resource files JDK-sef-methods.txt and omitmethods-defaults.txt, which
+ * contain a list of side effect free methods and a list of methods to omit during Randoop
+ * execution, respectively.
+ */
+public class MethodListGen {
+  /**
+   * The extension for a Java .class file. These files contain annotations that this tool will
+   * parse.
+   */
+  private static final String CLASS_EXT = ".class";
+
+  /** The type annotations indicating a non-deterministic return value. */
+  private static final Collection<String> NONDET_ANNOTATIONS =
+      Collections.singletonList("org.checkerframework.checker.determinism.qual.NonDet");
+
+  /** The method annotations indicating a side-effect-free method. */
+  private static final Collection<String> SEF_ANNOTATIONS =
+      Arrays.asList(
+          "org.checkerframework.dataflow.qual.SideEffectFree",
+          "org.checkerframework.dataflow.qual.Pure");
+
+  /** Text to place at the beginning of file {@code omitmethods-defaults.txt}. */
+  private static final String OMIT_METHODS_FILE_HEADER =
+      String.join(
+          System.lineSeparator(),
+          "# Long-running.  With sufficiently small arguments, can be fast.",
+          "# org.apache.commons.math3.analysis.differentiation.DSCompiler.getCompiler\\(int,int\\)",
+          "^org.apache.commons.math3.analysis.differentiation.",
+          "^org.apache.commons.math3.analysis.integration.",
+          "# Nondeterministic.");
+
+  /** An AnnotationLocation describes where to look for desired annotations. */
+  enum AnnotationLocation {
+    RETURN_TYPE,
+    METHOD
+  }
+
+  /**
+   * Main entry point to generate Nondeterministic and Side Effect Free Method lists.
+   *
+   * @param args command line arguments
+   *     <ul>
+   *       <li>args[0] - JDK .jar location as a path
+   *       <li>args[1] - output directory
+   *     </ul>
+   */
+  public static void main(String[] args) {
+    Path annotatedJar = Paths.get(args[0]);
+
+    Path nonDetFile = Paths.get(args[1], "omitmethods-defaults.txt");
+    Path sideEffectFreeFile = Paths.get(args[1], "JDK-sef-methods.txt");
+    Path unparsableSideEffectFreeFile = Paths.get(args[1], "JDK-sef-methods-unparsable.txt");
+
+    try {
+      List<String> nonDetMethods =
+          getAnnotatedMethodsFromJar(
+              annotatedJar, AnnotationLocation.RETURN_TYPE, NONDET_ANNOTATIONS);
+      List<String> sideEffectFreeMethods =
```

Comment:
Define variables as near their first use as possible.  This can come after the following if statement (near its own if statement).